src/System.Net.NameResolution/src/System/Net/DNS.cs

```diff
@@ -554,6 +571,9 @@ public static IPAddress[] GetHostAddresses(string hostNameOrAddress)
         public static IAsyncResult BeginGetHostEntry(string hostNameOrAddress, AsyncCallback requestCallback, object stateObject)
         {
             if (NetEventSource.Log.IsEnabled()) NetEventSource.Enter(NetEventSource.ComponentType.Socket, "DNS", "BeginGetHostEntry", hostNameOrAddress);
+
+            NameResolutionPal.EnsureSocketsAreInitialized();
```

Comment:
Also, please add some UnitTests to ensure that nobody removes the call by mistake - the Functional Tests may or may not catch based on order of execution.